src/mongocxx/exception/base.hpp

```diff
@@ -30,33 +27,50 @@ namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace exception {
 
-using error_and_code_type = std::tuple<std::string, std::int32_t>;
-
+///
+/// A class to be used as the base class for all mongocxx exceptions.
+///
 class MONGOCXX_API base : virtual public std::exception {
```

Comment:
So, here is an idea. We have been talking about unifying the space of bsoncxx and mongocxx errors by way of a common base class.

If we are committed to the idea that all exceptions that we throw will have a std::error_code member, why don't we just make that common base across the libraries be std::system_error?

We would need to make an error space for every error, but it would certainly simplify things. We wouldn't have a particularly interesting class hierarchy for the exceptions, but I've been thinking that that may be an anti-pattern to begin with.

If you made mongocxx::exception (maybe we drop the namespace) derive from std::system_error you wouldn't need to manage the error_code object or the what string yourself, only the optional raw_server_error. That seems like a big step forward.

The bsoncxx exception might just actually *be* a std::system_error, if there are no adornments that it requires.

@Machyne @amidvidy @hanumantmk @samantharitter - thoughts?